pontoon/projects/views.py

```diff
@@ -43,9 +44,23 @@ def projects(request):
 
 def project(request, slug):
     """Project dashboard."""
-    project = get_object_or_404(
-        Project.objects.visible_for(request.user).available(), slug=slug
-    )
+    try:
+        project = get_object_or_404(
+            Project.objects.visible_for(request.user).available(), slug=slug
+        )
+    except Http404:
+        # Try to find a project using old slug
+        slug_history = (
+            ProjectSlugHistory.objects.filter(old_slug=slug)
+            .order_by("-created_at")
+            .first()
+        )
+
+        if slug_history is not None:
+            # Redirect to the current project slug
+            return redirect("pontoon.projects.project", slug=slug_history.project.slug)
+        else:
+            raise Http404("Project not found.")
```

Comment:
There are quite a few views in which we'd need a check like this. For example, we'd also want to redirect the localization dashboard (/{locale}/{project}), translate view (/{locale}/{project}/all-resources/), as well as several AJAX views spread across multiple urls.py files.

Can we instead introduce a few URL patters that will catch old project slugs as proposed [initially](https://github.com/mozilla/pontoon/issues/2152#issuecomment-1591899132)?